gprofiler/metadata/application_identifiers.py

```diff
@@ -145,6 +145,53 @@ def get_app_id(self, process: Process) -> Optional[str]:
         return None
 
 
+class _UvicornApplicationIdentifierBase(_ApplicationIdentifier):
+    def uvicorn_to_app_id(self, wsgi_app_spec: str, process: Process) -> str:
+        wsgi_app_file = wsgi_app_spec.split(":", maxsplit=1)[0]
+        return f"uvicorn: {wsgi_app_spec} ({_append_python_module_to_proc_wd(process, wsgi_app_file)})"
+
+    def uvicorn_get_app_name(self, cmdline_list: List[str]) -> str:
+        # factory argument is being checked explicitly, because it contains minuses and precedes appid
```

Comment:
Btw where did you see this `--factory` occurring?

Reply:
https://www.uvicorn.org/#application-factories